File: 0161-One Edit Distance/0161-One Edit Distance.cpp
class Solution {
public:
    bool isOneEditDistance(string s, string t) {
        int m = s.size();
        int n = t.size();
        if (m > n) {
            return isOneEditDistance(t, s);
        }
        
        if (n - m > 1) {
            return false;
        }
        
        int diff = 0;
        if (m == n) {
            for (int i = 0; i < m; ++i) {
                if (s[i] != t[i]) {
                    ++diff;
                    if (diff > 1) {
                        return false;
                    }
                }
            }
            
            return diff == 1;
        }
        else {
            for (int i = 0, j = 0; i < m; ++i, ++j) {
                if (s[i] != t[j]) {
                    ++diff;
                    --i;
                    if (diff > 1) {
                        return false;
                    }
                }
            }
            
            return true;
        }
    }
};

File: 0101-0200/0161-One Edit Distance/0161-One Edit Distance.py
class Solution:
    def isOneEditDistance(self, s: str, t: str) -> bool:
        m, n = len(s), len(t)
        if m > n:
            m, n, s, t = n, m, t, s
        if n - m > 1:
            return False
        for i in range(m):
            if s[i] != t[i]:
                if m == n:
                    return s[i + 1:] == t[i + 1:]
                else:
                    return s[i:] == t[i+1:]
        return m + 1 == n
